src/platform/silabs/platformAbstraction/WiseMcuSpam.cpp

```diff
@@ -159,21 +159,20 @@ void sl_button_on_change(uint8_t btn, uint8_t btnAction)
     // Currently the btn0 is pull-up resistor due to which is sends a release event on every wakeup
     if (btn == SL_BUTTON_BTN0_NUMBER)
     {
-        if (btnAction == BUTTON_PRESSED)
+        // if the btn was not pressed and only a release event came, ignore it
+        // if the btn was already pressed and another press event came, ignore it
+        VerifyOrReturn(btnAction ^ btn0_pressed);
```

Comment:
As much as this is very net way of doing to the check, it is more readable to do it the "long" way. 
```suggestion
        VerifyOrReturn((btnAction != BUTTON_RELEASED) || (btn0_pressed != false));
```

Reply:
this suggested code does not work as expected for factory reset scenario.